PSMDB-1558 Release notes 6.0.19-16 (#947)
percona/psmdb-docs: docs/release_notes/6.0.19-16.md

# Percona Server for MongoDB 6.0.19-16 ({{date.6_0_19}})

[Installation](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB 6.0.19-16 is an enhanced, source-available, and highly-scalable database that is a
fully-compatible, drop-in replacement for MongoDB Community Edition. 

It is based on [MongoDB 6.0.19 Community Edition](https://www.mongodb.com/docs/manual/release-notes/6.0/#6.0.19---oct-24--2024) and supports the upstream protocols and drivers.

## Release Highlights

!!! important

>     This release of Percona Server for MongoDB includes a fix for a security vulnerability [CVE-2024-10921](https://nvd.nist.gov/vuln/detail/CVE-2024-10921). This vulnerability allowed an authorized user to trigger server crashes or receive the contents of the buffer over-reads of the server memory by sending specially crafted requests that constructed malfolmed BSON in MongoDB. The issue is fixed upstream in versions 6.0.18 and 7.0.15. The fix is also included in Percona Server for MongoDB  6.0.18-16 and 7.0.15-9.    

Reviewer: ```suggestion
    This release of Percona Server for MongoDB includes a fix for a security vulnerability [CVE-2024-10921](https://nvd.nist.gov/vuln/detail/CVE-2024-10921). This vulnerability allowed an authorized user to trigger server crashes or receive the contents of the buffer over-reads of the server memory by sending specially crafted requests that constructed malformed BSON in MongoDB. In case you want to upgrade to the following major version, the fix is also included in the Percona Server for MongoDB 7.0.15-9.    
```

>     This release of Percona Server for MongoDB includes a fix for a security vulnerability [CVE-2024-10921](https://nvd.nist.gov/vuln/detail/CVE-2024-10921). This vulnerability allowed an authorized user to trigger server crashes or receive the contents of the buffer over-reads of the server memory by sending specially crafted requests that constructed malformed BSON in MongoDB. The issue is fixed and included in Percona Server for MongoDB 6.0.18-16. If you wish to [upgrade to the following major version](../install/upgrade-from-50.md), the fix is also included in Percona Server for MongoDB 7.0.15-9.    

Reviewer 2: ```suggestion
    This release of Percona Server for MongoDB includes a fix for a security vulnerability [CVE-2024-10921](https://nvd.nist.gov/vuln/detail/CVE-2024-10921). This vulnerability allowed an authorized user to trigger server crashes or receive the contents of the buffer over-reads of the server memory by sending specially crafted requests that constructed malformed BSON in MongoDB. The issue is fixed and included in Percona Server for MongoDB 6.0.19-16. If you wish to [upgrade to the following major version](../install/upgrade-from-50.md), the fix is also included in Percona Server for MongoDB 7.0.15-9.    
```

>     Users running any minor version of Percona Server for MongoDB 6.0.x before 6.0.18-16 should upgrade to this version as soon as possible. 

Reviewer 2: ```suggestion
    Users running any minor version of Percona Server for MongoDB 6.0.x before 6.0.19-16 should upgrade to this version as soon as possible. 
```

>     Users running any minor version of Percona Server for MongoDB 6.0.x before 6.0.18 and Percona Server for MongoDB 7.0.x before 7.0.15 should upgrade to Percona Server for MongoDB 6.0.18-16 and 7.0.15-9, respectively, as soon as possible. 

Reviewer: ```suggestion
    Users running any minor version of Percona Server for MongoDB 6.0.x before 6.0.18-16 should upgrade to this version as soon as possible. 
```

### Upstream Improvements

The bug fixes, provided by MongoDB and included in Percona Server for MongoDB, are the following:

* [SERVER-96419](https://jira.mongodb.org/browse/SERVER-96419) - Fixed the issue with improper neutralization of null bytes that may have led to buffer over-reads in MongoDB Server.
* [SERVER-95279](https://jira.mongodb.org/browse/SERVER-95279) - Use a new C++ type for BSON field names to ensure validity.

Find the full list of changes in the [MongoDB 6.0.19 Community Edition release notes](https://www.mongodb.com/docs/manual/release-notes/6.0/#6.0.19---oct-24--2024).

## Changelog

### Improvements

> * [PSMDB-1539](https://perconadev.atlassian.net/browse/PSMDB-1539) - Removed the fork:true and pidfile management to simplify migration.

Reviewer 2: ```suggestion
```

> * [PSMDB-1539](https://perconadev.atlassian.net/browse/PSMDB-1539) - Removed the fork:true and pidfile management to simplify migration.

Reviewer: ```suggestion
* [PSMDB-1539](https://perconadev.atlassian.net/browse/PSMDB-1539) - Removed the fork:true and pidfile management to simplify migration from the upstream versions.  
```
